src/net.cpp

```diff
@@ -718,6 +720,127 @@ CNetMessage V1TransportDeserializer::GetMessage(const CMessageHeader::MessageSta
     return msg;
 }
 
+int V2TransportDeserializer::Read(const char* pch, unsigned int bytes)
+{
+    if (!m_in_data) {
+        // copy data to temporary parsing buffer
+        const unsigned int remaining = CHACHA20_POLY1305_AEAD_AAD_LEN - m_hdr_pos;
+        const unsigned int copy_bytes = std::min(remaining, bytes);
+
+        memcpy(&vRecv[m_hdr_pos], pch, copy_bytes);
+        m_hdr_pos += copy_bytes;
+
+        // if AAD incomplete, exit
+        if (m_hdr_pos < CHACHA20_POLY1305_AEAD_AAD_LEN) {
```

Comment:
Why the encrypted length doesn't get its own MAC to guarantee integrity ? It can decrypt to some garbage and therefore open to manipulating ciphertext-MAC seek in message ?

Reply:
BIP discussion. The length is covered by the MAC.